src/Extension/ExtendableTrait.php

```diff
@@ -225,7 +226,50 @@ public function extendClassWith($extensionName)
     public function isClassExtendedWith($name)
     {
         $name = str_replace('.', '\\', trim($name));
-        return isset($this->extensionData['extensions'][$name]);
+
+        if (isset($this->extensionData['extensions'][$name])) {
+            return true;
+        }
+
+        $parts = explode('\\', $name);
+        $count = count($parts) - 1;
+
+        if ($count < 1) {
+            return false;
+        }
+
+        $map = [];
+        $namespace = '';
+
+        for ($i = 0; $i < $count; $i++) {
+            $namespace = $namespace ? $namespace . '\\' . $parts[$i] : $parts[$i];
+            $map[] = $namespace;
+        }
+
+        $map = array_reverse($map);
+
+        foreach ($map as $namespace) {
+            if ($aliases = $this->getClassLoader()->getNamespaceAliases($namespace)) {
+                foreach ($aliases as $alias) {
+                    if (isset($this->extensionData['extensions'][str_replace($namespace, $alias, $name)])) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
```

Comment:
Can't we just do 
```php
return isset($this->extensionData['extensions'][$name]) || isset($this->extensionData['extensions'][$classLoader->getAlias($name)])
```

Reply:
Not really, the aliases returned are an array as you can have multiple aliases registered pointing back to one original. This also differs from class aliasing as it's an alias of a segment of the namespace, so we need to break down the namespace of `$name` and check if any part of it has an alias that is being used for the implement. Sorry if that isn't clear, it's 4am :) lol

Reply:
So is the extensionData array populated with the class names as they were when they were registered instead of the real class name of the extension? If that's the case, could we not resolve the class alias using getAlias() when the extension is first registered, and then always resolve aliases when checking for existence?

Reply:
Yes to the first question, not sure about the second, I don't think there was an alias set for the implement class specifically as it was covered by the namespace alias registered during the `PluginManager` init. So we have to more or less guess at the namespace. I could take a look tomorrow though and see what we can do

Reply:
getAlias() resolves namespace aliases FYI. Tomorrow sounds good though, get some sleep already! 💤

Reply:
![image](https://user-images.githubusercontent.com/31214002/116692069-b1abd280-a9b3-11eb-95ca-9f325966a8bc.png)

So `getAlias` will resolve an alias to the original, but what we actually want to check is if a class has an alias and if that alias is being used. 

I.e.
- Find all aliases for the namespace being used
- Check each to see if it is being implmented

Reply:
> what we actually want to check is if a class has an alias and if that alias is being used.
Why do we need to check that?

Reply:
So for instance, if the user has installed:
- Winter.Translate
- Offline.Mall

Mall has a `Product` model that implement `RainLab.Translate.Behaviors.TranslatableModel`. Which is handled by the class loader when the model gets extended, but `ExtendableTrait::isClassExtendedWith()` will return false if you pass it `Winter.Translate.Behaviors.TranslatableModel` because it's not in the `extensionData` array.

So when calling `isClassExtendedWith()` we want to check if the object has been extended with an alias of a original class.

Reply:
Another option might be making changes here:

![image](https://user-images.githubusercontent.com/31214002/116720698-6ce46380-a9d4-11eb-8a2e-9bac26ca38c9.png)

And modifying `$extensionName` to be the actual name of object created